src/System.Data.Common/src/System/Data/Common/DbConnection.cs

```diff
@@ -35,6 +35,8 @@ protected DbConnection() : base()
         /// </summary>
         protected virtual DbProviderFactory DbProviderFactory => null;
 
+        internal DbProviderFactory ProviderFactory=>DbProviderFactory;
```

Comment:
I'll add a test for this tomorrow as well, next to the test in the other issue and will add a comment why it's there.